src/store/auth.js

```diff
@@ -1,64 +1,98 @@
+    if (!result.credential) return null;
+    return result.user;
   } catch (err) {
-    console.error('getRedirectResult error', {
+    console.error("getRedirectResult error", {
       code: err.code,
       message: err.message,
       email: err.email,
       credential: err.credential
-    })
+    });
   }
-}
+};
 
 export const state = () => {
   return {
     user: null
-  }
-}
+  };
+};
 
 export const mutations = {
-  setUser (state, userData) {
-    const fields = ['uid', 'displayName', 'email', 'emailVerified', 'photoURL']
-    const user = _.pick(userData, fields)
-    state.user = user
+  setUser(state, userData) {
+    const fields = ["uid", "displayName", "email", "emailVerified", "photoURL"];
+    const user = _.pick(userData, fields);
+    state.user = user;
   }
-}
+};
 
 export const actions = {
-  async init ({ commit }) {
-    const redirectUser = await getRedirectResult(this.$fire)
-    if (redirectUser) commit('setUser', redirectUser)
+  async init({ commit }) {
+    const redirectUser = await getRedirectResult(this.$fire);
+    if (redirectUser) commit("setUser", redirectUser);
   },
-  signInWithRedirect (ctx, provider) {
+  signInWithRedirect(ctx, provider) {
     switch (_.toLower(provider)) {
-      case 'github':
-        this.$fire.auth.signInWithRedirect(new this.$fireModule.auth.GithubAuthProvider())
-        break
-      case 'google':
-        this.$fire.auth.signInWithRedirect(new this.$fireModule.auth.GoogleAuthProvider())
-        break
-      case 'facebook':
-        this.$fire.auth.signInWithRedirect(new this.$fireModule.auth.FacebookAuthProvider())
-        break
-      case 'twitter':
-        this.$fire.auth.signInWithRedirect(new this.$fireModule.auth.TwitterAuthProvider())
-        break
+      case "github":
+        this.$fire.auth.signInWithRedirect(
+          new this.$fireModule.auth.GithubAuthProvider()
+        );
+        break;
+      case "google":
+        this.$fire.auth.signInWithRedirect(
+          new this.$fireModule.auth.GoogleAuthProvider()
+        );
+        break;
+      case "facebook":
+        this.$fire.auth.signInWithRedirect(
+          new this.$fireModule.auth.FacebookAuthProvider()
+        );
+        break;
+      case "twitter":
+        this.$fire.auth.signInWithRedirect(
+          new this.$fireModule.auth.TwitterAuthProvider()
+        );
+        break;
     }
   },
-  onAuthStateChange ({ commit }, { authUser, claims }) {
+  onAuthStateChanged({ commit }, { authUser, claims }) {
     if (!authUser) {
-      this.$fire.auth.signOut()
-      commit('setUser', null)
-      return
+      this.$fire.auth.signOut();
+      commit("setUser", null);
+      return;
     }
-    
-    commit('setUser', authUser)
+
+    commit("setUser", authUser);
+  },
+  signOut() {
+    this.$fire.auth.signOut().then(
+      function() {
+        console.log("Signed Out");
+      },
+      function(error) {
+        console.error("Sign Out Error", error);
+      }
+    );
+  },
```

Comment:
Huh... I can't believe I never had a method for this despite importing it on the index page.

It's not wrong, but it's also not pretty. We're not using the `.then` or `.catch` style promises in this app, and generally speaking I avoid them whenever possible especially now that [top-level await](https://v8.dev/features/top-level-await) is becoming more prevalent.

Also, I think this might be the first time I've ever seen someone use the second argument to `.then()` to catch the error. Like I said, it's not wrong, but that takes precedence over `.catch` and 99 times out of 100 you'll promises in this style written in the `.then().catch()` format. That said, that format is also ugly and async/await is best for readability and code flow.

Reply:
That reminds me, we actually don't care if this is successful: the onAuthStateChanged handler will update the store's state for us. But we do care if there was an error. So let's still log out that `console.error` you have, but can you change the second argument to it to be `error.message` instead? I typically find that's all I really care about in debugging and having the entire error object logged in the console is unnecessary.

Reply:
Updated with try/catch to accomodate being able to receive error.message if error on login. Upon researching top-level await didn't know if wanted to change auth.js store to auth.mjs file type or import "type":"module" across whole project in package.json. Happy to speak more about proper implementation. Found good resource here : https://www.stefanjudis.com/today-i-learned/top-level-await-is-available-in-node-js-modules/